numpydoc/tests/test_numpydoc.py

```diff
@@ -64,6 +64,20 @@ def test_mangle_docstrings():
     assert 'upper' not in [x.strip() for x in lines]
 
 
+def test_clean_text_signature():
+    assert _clean_text_signature(None) is None
+    assert _clean_text_signature('func($self)') == 'func()'
+    assert (_clean_text_signature('func($self, *args, **kwargs)')
+            == 'func(*args, **kwargs)')
+    assert _clean_text_signature('($self)') == '()'
+    assert _clean_text_signature('()') == '()'
+    assert _clean_text_signature('func()') == 'func()'
+    assert (_clean_text_signature('func($self, /, *args, **kwargs)')
+            == 'func(*args, **kwargs)')
+    assert _clean_text_signature('($module)') == '()'
+    assert _clean_text_signature('func($type)') == 'func()'
+
```

Comment:
please add a test where there are other params with default values, such as
```
_clean_text_signature('func($self, foo="hello world")')
_clean_text_signature('func(foo="hello world")')
_clean_text_signature('func(foo="$self")')
```
and tricky cases like
```
_clean_text_signature('func(self, other)')
```

Reply:
done